weldx/util/util.py

```diff
@@ -271,7 +277,9 @@ def compare_nested(
         """Deeply compares [nested] data structures combined of tuples, lists, dicts...
 
         Also compares non-nested data-structures.
-        Arrays are compared using np.all and xr.DataArray.identical
+        Arrays are compared using np.all and xr.DataArray.identical.
+        Sets are compared by calling ``==`` and not traversed, hence the order of all
+        set items is important.
```

Comment:
How is the order important or which order do you mean? `{"a", "b"} == {"b", "a"}` is `True`.

Reply:
The order only really becomes important if the set contains some complex/nested elements (basically when you cannot apply sorting to the set items), see the `test_nested_sets` test case example